src/transformers/models/bert/modeling_tf_bert.py

```diff
@@ -314,31 +214,28 @@ def __init__(self, config: BertConfig, **kwargs):
                 f"of attention heads ({config.num_attention_heads})"
             )
 
+        self.num_attention_heads = config.num_attention_heads
         self.attention_head_size = int(config.hidden_size / config.num_attention_heads)
+        self.all_head_size = self.num_attention_heads * self.attention_head_size
+        self.rsqrt_att_head_size = tf.cast(1.0 / math.sqrt(self.attention_head_size), dtype=self._compute_dtype)
```

Comment:
No I don't think this will work -> the dtype could change if the user changes their policy after instantiating the model. We should remove the `dtype = self._compute_dtype` and do the `cast` in the call as was done before.
(Relying on the [mixed precision doc](https://www.tensorflow.org/api_docs/python/tf/keras/mixed_precision/Policy#how_to_write_a_layer_that_supports_mixed_precision_and_float64_2) here.)

Reply:
This is what I answered in the Morgan's comment https://github.com/huggingface/transformers/pull/9890#discussion_r567723981

Reply:
Yes, and I disagree with it. This is not how the documentation indicate we should treat new tensors values and we should not assume the user is always setting the right policy before instantiating their model.